### DISCUSSION TITLE
Loading custom objects?

### DISCUSSION THREAD

**Discussion by sidgairo18** [rafael-fuente/Python-Raytracer#8] [Q&A] [answered]
I have an object .stl file which I have build using a 3D CAD software. Can that be imported into the scene?
Or is there any other library where such objects can be imported and rendered.

Thanks.

**Comment by rafael-fuente** [comment, marked as answer]
I have implemented a class named  [TriangleMesh](https://github.com/rafael-fuente/Python-Raytracer/blob/master/sightpy/geometry/triangle_mesh.py)  to load arbitrary .obj files as triangle meshes. So, first convert your .stl file to a .obj format.

However, the rendering is quite slow because we need to loop for every triangle of the mesh, and Python loops are very slow. A [bounding volume hierarchy acceleration structure](https://www.scratchapixel.com/lessons/advanced-rendering/introduction-acceleration-structure/bounding-volume-hierarchy-BVH-part1) could be implemented, but I think renders will still take a lot of time.

If you want to render arbitrary triangle meshes, Python isn't the language to work with. An implementation in a compiled language like C++ should be needed for speed.
